### ISSUE TITLE
Refactor Developing Applications documentation

### ISSUE THREAD

**Issue by denyeart** [hyperledger/fabric#2983]
"Developing Applications" [doc](https://hyperledger-fabric.readthedocs.io/en/latest/developapps/developing_applications.html) has gotten somewhat outdated and is based on the Commercial Paper chaincode that many people don't understand. It should be overhauled and potentially based on a simpler chaincode that new developers can more easily understand.

Additionally it needs to be refactored for the new embedded peer Gateway.

**Comment by DARK-art108** [comment]
Can I work on this issue? @denyeart

**Comment by denyeart** [comment]
@DARK-art108 Yes, please coordinate with @bestbeforetoday (aka Mark Lewis, aka bestbeforetoday) on the proposed approach. Mark is the application SDK lead and should be involved in Developing Applications changes.

**Comment by jonmchan** [comment]
Offline signing seems like a significant feature or use case for the Fabric Gateway SDKs. I was unable to find it in the documentation anywhere except in the code and tests. Would this be something that should get a section?

**Comment by bestbeforetoday** [comment]
> Offline signing seems like a significant feature or use case for the Fabric Gateway SDKs. I was unable to find it in the documentation anywhere except in the code and tests. Would this be something that should get a section?

There is some description and code examples in the API documentation for the Contract in each of the client languages:

- Go: https://pkg.go.dev/github.com/hyperledger/fabric-gateway/pkg/client#Contract
- Node: https://hyperledger.github.io/fabric-gateway/main/api/node/interfaces/Contract.html
- Java: https://hyperledger.github.io/fabric-gateway/main/api/java/org/hyperledger/fabric/client/Contract.html

Any additional documentation giving a language agnostic overview of the client API structure and usage patterns (perhaps a reworked developing applications section) should cover off-line signing too.

**Comment by denyeart** [comment]
Developing Applications doc has been removed. Intent now is to add some of the main concepts to the Key Concepts and Tutorial topics.

**Comment by joaormatos** [comment]
> Developing Applications doc has been removed. Intent now is to add some of the main concepts to the Key Concepts and Tutorial topics.

Meanwhile there are a few dangling links to the Developing Applications section in the current version of the documentation.

[A code search for _developapps_ easily reveals them.](https://github.com/search?q=repo%3Ahyperledger%2Ffabric+path%3A%2F%5Edocs%5C%2F%2F+developapps&type=code)

As a result, I couldn't find a full explanation of chaincode namespaces in the current version of the docs.
Maybe it's in there and I didn't look hard enough, but I personally ended up hunting down and consulting the deleted section in version 2.3 of the docs to learn about namespaces.

Here's a helpful link to the old docs in case anyone else is looking for this and runs into this comment.
https://hyperledger-fabric.readthedocs.io/en/release-2.3/developapps/chaincodenamespace.html

**Comment by denyeart** [comment]
The broken links to the old Developing Applications doc topics need to be removed.

**Comment by kiranmenon** [comment]
reached here after getting 404 for https://hyperledger-fabric.readthedocs.io/en/release-2.5/developapps/developing_applications.html from the readthedocs.

**Comment by ctsatheesh** [comment]
Where is the link for cross chaincode (chaincode namespace) section ?

In release 2.3, it was found here:
https://hyperledger-fabric.readthedocs.io/en/release-2.3/developapps/chaincodenamespace.html#cross-chaincode-access

In release 2.5, what is the corresponding section that talks about invoking chaincode in other channels ? 
Are the restrictions of invoking smart contracts to update world state of called contract still applies ?

**Comment by bestbeforetoday** [comment]
When the [Fabric Gateway client API](https://hyperledger.github.io/fabric-gateway/) replaced the various legacy SDKs for developing client applications with Fabric v2.5, there was not enough developer bandwidth to update the [Commercial Paper](https://github.com/hyperledger/fabric-samples/tree/release-2.2/commercial-paper) sample and accompanying [Developing Applications](https://hyperledger-fabric.readthedocs.io/en/release-2.2/developapps/developing_applications.html) documentation. Instead, the [Running a Fabric Application](https://hyperledger-fabric.readthedocs.io/en/release-2.5/write_first_app.html) tutorial was re-written for the new client API and the Developing Application section was removed.

If somebody has the time to invest in rewriting the Commercial Paper sample and all the Developing Application content, I'm sure that would be a welcome addition to the current documentation.

The guidance and restrictions on cross chaincode invocation from the old Developing Applications documentation is still valid.
